planning/obstacle_cruise_planner/src/optimization_based_planner/velocity_optimizer.cpp

```diff
@@ -252,25 +252,29 @@ VelocityOptimizer::OptimizationResult VelocityOptimizer::optimize(const Optimiza
     std::cerr << "optimization failed : " << qp_solver_.getStatusMessage().c_str() << std::endl;
   }
 
-  std::vector<double> opt_time = time_vec;
-  std::vector<double> opt_pos(N);
-  std::vector<double> opt_vel(N);
-  std::vector<double> opt_acc(N);
-  std::vector<double> opt_jerk(N);
-  for (size_t i = 0; i < N; ++i) {
-    opt_pos.at(i) = optval.at(IDX_S0 + i);
-    opt_vel.at(i) = std::max(optval.at(IDX_V0 + i), 0.0);
-    opt_acc.at(i) = optval.at(IDX_A0 + i);
-    opt_jerk.at(i) = optval.at(IDX_J0 + i);
-  }
-  opt_vel.back() = 0.0;
-
   OptimizationResult optimized_result;
-  optimized_result.t = opt_time;
-  optimized_result.s = opt_pos;
-  optimized_result.v = opt_vel;
-  optimized_result.a = opt_acc;
-  optimized_result.j = opt_jerk;
+  const auto has_nan =
+    std::any_of(optval.begin(), optval.end(), [](const auto v) { return std::isnan(v); });
+  if (!has_nan) {
```

Comment:
It is just a matter of style, but we could also simply return early here:
```C++
if (has_nan) {
  // print err
  return {};
}
// fill and return output
```

Reply:
Thank you for your comment. I was trying to avoid having multiple return statements which is why I went with the current style.